### DISCUSSION TITLE
Release v1.0.18 / 5.55.18

### DISCUSSION THREAD

**Discussion by DavidXanatos** [sandboxie-plus/Sandboxie#1793] [Announcements]
![0 5 5](https://user-images.githubusercontent.com/3890945/105014988-575cb680-5a41-11eb-90d5-d942d28184b2.png)

This build fixes a couple of issues, but also introduces a major change in how sandboxie controls access to process memory.

Before this build sandboxie allowed sandboxed programs to read the memory of any unsandboxed program belonging to the current user, this is obviously a bad idea if your goals is not only infection prevention but also data protection. Hence with 1.0.16 onwards sandboxie will not allow for PROCESS_VM_READ on unsandboxed processes or processes belonging to other boxes.
To facilitate compatibility this build introduces a IPC options, with ReadIpcPath=$:program.exe any unboxed process can be configured to allow for PROCESS_VM_READ, it is also possible to restore the old behavior entirely by specifying ReadIpcPath=$:* 
By default the only process whos memory can be read is explorer.exe many processes want that and explorer should not keep any secrets normally anyways. To block this you can use ClosedIpcPath=$:explorer.exe

To facilitate optimal process isoaltion the EnableObjectFiltering option is now on by default, although this only applies for new installations, hence its recommend for existing installation to go to settings->advanced and enable it explicitly.

Other changes in this build include a simple resource access monitor mode and a change how process paths are resolved for sandboxed processes, this should fix a couple of issues.

Given that this build changes a couple of core mechanics it is possible that in some special cases this can lead to an incompatibility.

#### You can support the project through donations, any help will be greatly appreciated.

[<img src="https://www.paypalobjects.com/en_US/AT/i/btn/btn_donateCC_LG.gif">](https://xanasoft.com/go.php?to=donate)

# ChangeLog

### Added
- added minor browsers to BlockSoftwareUpdaters template (by APMichael) [#1784](https://github.com/sandboxie-plus/Sandboxie/pull/1784)

### Changed
- Failed memory read attempts to unboxed processes will no longer cause message 2111 by default
-- Note: the message can be enabled in the settings if desired with "NotifyProcessAccessDenied=y"
- reordered BlockSoftwareUpdaters template (by APMichael) [#1785](https://github.com/sandboxie-plus/Sandboxie/pull/1785)

### Fixed
- fixed pipe impersonation in compartment mode
- fixed issue with box clean-up introduced in a recent build
- fixed missing trace log cleanup command [#1773](https://github.com/sandboxie-plus/Sandboxie/issues/1773)
- fixed unpin did not work [#1694](https://github.com/sandboxie-plus/Sandboxie/issues/1694)

### Added
- added checkbox for easy read access to memory of unsandboxed processes (old Sbie behaviour, not recommended)

### Changed
- improved OpenProcess/OpenThread logging

### Fixed
- fixed crash issue with the new monitor mode
- fixed issue with resource access entry parsing

### Added
- FIXED SECURITY ISSUE: memory of unsandboxed processes can no longer be read, exceptions are possible
-- you can use ReadIpcPath=$:program.exe to allow read access to unsandboxed processes or processes in other boxes
- Added "Monitor Mode" to the resource access trace, similar to the old monitor view of SbieCtrl.exe

### Changed
- EnableObjectFiltering is now set enabled by default, and replaces Sbie's old process/thread handle filter
- the `$:` syntax now accepts a wildcard `$:*` no more specialized wildcards though

### fixed
- fixed NtGetNextProcess being fully disabled instead of properly filtered
- fixed reworked image name resolution when creating new processes in a sandbox
- fixed regression with HideOtherBoxes=y [#1743](https://github.com/sandboxie-plus/Sandboxie/issues/1743) [#1666](https://github.com/sandboxie-plus/Sandboxie/issues/1666)

<hr /><em>This discussion was created from the release <a href='https://github.com/sandboxie-plus/Sandboxie/releases/tag/1.0.18'>Release v1.0.18 / 5.55.18</a>.</em>

**Comment by BilleBarrett** [comment]
installed on 4 computers 1 Sandboxie & 3 Sandbox-Plus 
Everything is running OK

**Reply by BilleBarrett** [reply]
Ok I'm releasing it to my clients and looking good, only 200 more to go

**Comment by BilleBarrett** [comment]
can someone remind me what has to be set when you click on a link in outlook and it fails to go there
Thank you
Browser is Firefox is his browser
It turned out after looking at the default browser the client installed AVG browser protector, uninstall and it works not
Sorry to bug you guys
